src/core/Str.pm

```diff
@@ -1058,8 +1076,12 @@ my class Str does Stringy { # declared in BOOTSTRAP
         )
     }
 
+    proto method subst-mutate(|) {
+        $/ := nqp::getlexdyn('$/');
+        {*}
+    }
     multi method subst-mutate(
-      Str:D $self is rw: $matcher, $replacement,
+      Str:D $self is rw: Any:D $matcher, Cool(Str:D) $replacement,
```

Comment:
The `Cool(Str:D)` seems to be wrong ("convert from Str to Cool"). But the type-constraint itself here is wrong, as the `$replacement` can also be a `Callable`. If our spectests don't catch that, we need to add such tests.